presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorFactory.java

```diff
@@ -19,6 +19,10 @@
 {
     String getName();
 
+    /**
+     * Gets the default handle resolver for connectors created by this factory.  If this return null,
+     * {@link Connector#getHandleResolver()} must not return an empty result.
+     */
     ConnectorHandleResolver getHandleResolver();
```

Comment:
Maybe we should deprecate this method.

Reply:
@kokosing I am not sure I want deprecation here.  Let consider this separately, as a follow-up.